frontend/src/components/resourceMap/GraphView.tsx

```diff
@@ -0,0 +1,388 @@
+    () => ({ nodeSelection: selectedNodeId, highlights, setNodeSelection: setSelectedNodeId }),
+    [selectedNodeId, setSelectedNodeId, highlights]
+  );
+  return (
+    <GraphViewContext.Provider value={contextValue}>
+      <Box
+        sx={{
+          position: 'relative',
+          height: height ?? '800px',
+          display: 'flex',
+          flexDirection: 'row',
+          flex: 1,
+        }}
+      >
+        <CustomThemeProvider>
+          <Box
+            sx={{
+              overflow: 'hidden',
+              display: 'flex',
+              flexDirection: 'column',
+              position: 'relative',
+              flexGrow: 1,
+              background: '#00000002',
+            }}
+          >
+            <Box
+              padding={2}
+              pb={0}
+              display="flex"
+              gap={1}
+              alignItems="center"
+              mb={1}
+              flexWrap="wrap"
+            >
+              <ResourceSearch
+                resources={nodes.flatMap(it =>
+                  it.type === 'kubeObject' ? [it.data.resource] : []
+                )}
+                onSearch={resource => {
+                  setSelectedNodeId(resource.metadata.uid);
+                }}
+              />
+
+              <NamespacesAutocomplete />
+
+              <GraphSourcesView
+                sources={defaultSources}
+                selectedSources={selectedSources}
+                toggleSource={toggleSelection}
+                sourceData={sourceData ?? new Map()}
+              />
+
+              {namespaces.size !== 1 && (
+                <ChipToggleButton
+                  label={t('Group By: {{ name }}', { name: t('Namespace') })}
+                  isActive={groupBy === 'namespace'}
+                  onClick={() => setGroupBy(it => (it === 'namespace' ? undefined : 'namespace'))}
+                />
+              )}
+
+              <ChipToggleButton
+                label={t('Group By: {{ name }}', { name: t('Instance') })}
+                isActive={groupBy === 'instance'}
+                onClick={() => setGroupBy(it => (it === 'instance' ? undefined : 'instance'))}
+              />
+
+              <ChipToggleButton
+                label={t('Group By: {{ name }}', { name: t('Node') })}
+                isActive={groupBy === 'node'}
+                onClick={() => setGroupBy(it => (it === 'node' ? undefined : 'node'))}
+              />
+
+              <ChipToggleButton
+                label={t('Status: Error or Warning')}
+                isActive={hasErrorsFilter}
+                onClick={() => setHasErrorsFilter(!hasErrorsFilter)}
+              />
+
+              {graphSize < 50 && (
+                <ChipToggleButton
+                  label={t('Expand All')}
+                  isActive={expandAll}
+                  onClick={() => setExpandAll(it => !it)}
+                />
+              )}
+            </Box>
+
+            <div style={{ flexGrow: 1 }}>
```

Comment:
html tags shouldn't be used

Reply:
why not? rendering a bunch of Boxes components noticeably affect performance here

Reply:
This is a nit, you can ignore if you want.

> why not? 

For consistency.

> rendering a bunch of Box components noticeably affects performance here

There's several react components at the same level, so I'm skeptical if this is true.

Reply:
Let's skip this.